File: app/src/main/java/com/mirogal/cocktail/data/repository/netpagedlist/DataSourceFactory.java
package com.mirogal.cocktail.data.repository.netpagedlist;

import androidx.lifecycle.MutableLiveData;

import com.mirogal.cocktail.data.database.entity.CocktailDbEntity;

public class DataSourceFactory extends androidx.paging.DataSource.Factory<Integer, CocktailDbEntity> {

    // TODO Dagger integration

    private final MutableLiveData<DataSource> mutableLiveData = new MutableLiveData<>();

    private DataSource cocktailDataSource;
    private String currentQuery;

    public void setCurrentQuery(String currentQuery) {
        this.currentQuery = currentQuery;
    }

    @Override
    public DataSource create() {
        cocktailDataSource = new DataSource(currentQuery);
        mutableLiveData.postValue(cocktailDataSource);
        return cocktailDataSource;
    }
}

File: app/src/main/java/com/mirogal/cocktail/data/repository/netpagedlist/DataSourceFactory.kt
package com.mirogal.cocktail.data.repository.netpagedlist

import androidx.lifecycle.MutableLiveData
import com.mirogal.cocktail.data.db.entity.CocktailDbEntity

class DataSourceFactory : androidx.paging.DataSource.Factory<Int?, CocktailDbEntity?>() {

    private val mutableLiveData = MutableLiveData<DataSource>()
    private var currentQuery: String? = null

    fun setCurrentQuery(currentQuery: String?) {
        this.currentQuery = currentQuery
    }

    override fun create(): DataSource {
        val cocktailDataSource = DataSource(currentQuery)
        mutableLiveData.postValue(cocktailDataSource)
        return cocktailDataSource
    }

}